Lib/test/test_sample_profiler.py

```diff
@@ -0,0 +1,1817 @@
+            # Suppress profiler output
+            io.StringIO() as captured_output,
+            mock.patch("sys.stdout", captured_output),
+        ):
+            try:
+                profile.sample.sample(
+                    proc.pid,
+                    duration_sec=1,
+                    all_threads=True,
+                    sample_interval_usec=10000,
+                    show_summary=False,
+                )
+            except PermissionError:
+                self.skipTest("Insufficient permissions for remote profiling")
+
+        # Just verify that sampling completed without error
+        # We're not testing output format here
+
+
+@unittest.skipIf(
+    sys.platform == "linux" and not PROCESS_VM_READV_SUPPORTED,
+    "Test only runs on Linux with process_vm_readv support",
+)
+class TestSampleProfilerErrorHandling(unittest.TestCase):
+    def test_invalid_pid(self):
+        with self.assertRaises((OSError, RuntimeError)):
+            profile.sample.sample(-1, duration_sec=1)
+
+    def test_process_dies_during_sampling(self):
+        with test_subprocess("import time; time.sleep(0.5); exit()") as proc:
+            with (
+                io.StringIO() as captured_output,
+                mock.patch("sys.stdout", captured_output),
+            ):
+                try:
+                    profile.sample.sample(
+                        proc.pid,
+                        duration_sec=2,  # Longer than process lifetime
+                        sample_interval_usec=50000,
+                    )
+                except PermissionError:
+                    self.skipTest(
+                        "Insufficient permissions for remote profiling"
+                    )
+
+                output = captured_output.getvalue()
+
+            self.assertIn("Error rate", output)
+
+    def test_invalid_output_format(self):
+        with self.assertRaises(ValueError):
+            profile.sample.sample(
+                os.getpid(),
+                duration_sec=1,
+                output_format="invalid_format",
+            )
+
+    def test_invalid_output_format_with_mocked_profiler(self):
+        """Test invalid output format with proper mocking to avoid permission issues."""
+        with mock.patch(
+            "profile.sample.SampleProfiler"
+        ) as mock_profiler_class:
+            mock_profiler = mock.MagicMock()
+            mock_profiler_class.return_value = mock_profiler
+
+            with self.assertRaises(ValueError) as cm:
+                profile.sample.sample(
+                    12345,
+                    duration_sec=1,
+                    output_format="unknown_format",
+                )
+
+            # Should raise ValueError with the invalid format name
+            self.assertIn(
+                "Invalid output format: unknown_format", str(cm.exception)
+            )
+
+    def test_is_process_running(self):
+        with (test_subprocess("import time; time.sleep(0.2)") as proc,
```

Comment:
This is going to race. The only reason it works is because seems that `_is_process_running` returns True for zombie processes :sweat: